app.py

```diff
@@ -23,10 +24,17 @@
 def static_file(path):
     return app.send_static_file(path)
 
+USE_RBAC = False
+
+if os.environ.get("AUTH_TYPE") == 'rbac':
+    USE_RBAC = True
+    credential = DefaultAzureCredential()
+    openai_token = credential.get_token("https://cognitiveservices.azure.com/.default")
```

Comment:
Looks like this also uses old openAI SDK with tokens that must be manually refreshed. Recommend moving to new SDK to take care of token refresh for you.

Reply:
Upgrade `openai` to V1.